db/migrate/20160113195732_copy_project_city_from_project_account.rb

```diff
@@ -0,0 +1,31 @@
+class CopyProjectCityFromProjectAccount < ActiveRecord::Migration
+  def up
+    execute <<-SQL
+-- Old data won't be validated here
+ALTER TABLE projects DISABLE TRIGGER sent_validation;
+
+UPDATE projects
+SET city_id = (
+    SELECT min(c.id)
+    FROM
+        cities c
+        JOIN states s ON s.id = c.state_id
+        JOIN project_accounts pa ON lower(pa.address_state) = lower(s.acronym) AND lower(unaccent(pa.address_city)) = lower(unaccent(c.name))
+    WHERE pa.project_id = projects.id
+)
+WHERE
+    city_id IS NULL
+    AND EXISTS (
+        SELECT true
+        FROM project_accounts pa
+        WHERE pa.project_id = projects.id
+    )
+    AND projects.state_order >= 'published';
```

Comment:
Shouldn't this start from in_analysis since that is when the city becomes required?